pkg/operator/ceph/cluster/mgr/dashboard.go

```diff
@@ -110,58 +105,72 @@ func (c *Cluster) configureDashboardModules() error {
 
-	daemonID = fmt.Sprintf("mgr.%s", daemonID)
+	monStore := config.GetMonStore(c.context, c.clusterInfo)
 
 	// url prefix
-	hasChanged, err := monStore.SetIfChanged(daemonID, "mgr/dashboard/url_prefix", c.spec.Dashboard.URLPrefix)
+	hasChanged, err := monStore.SetIfChanged(config.MgrType, "mgr/dashboard/url_prefix", c.spec.Dashboard.URLPrefix)
 	if err != nil {
 		return false, err
 	}
 
 	// ssl support
 	ssl := strconv.FormatBool(c.spec.Dashboard.SSL)
-	changed, err := monStore.SetIfChanged(daemonID, "mgr/dashboard/ssl", ssl)
+	changed, err := monStore.SetIfChanged(config.MgrType, "mgr/dashboard/ssl", ssl)
 	if err != nil {
 		return false, err
 	}
 	hasChanged = hasChanged || changed
 
 	// Prometheus host end point
 	prometheusEndpoint := c.spec.Dashboard.PrometheusEndpoint
-	changed, err = monStore.SetIfChanged(daemonID, "mgr/dashboard/PROMETHEUS_API_HOST", prometheusEndpoint)
+	changed, err = monStore.SetIfChanged(config.MgrType, "mgr/dashboard/PROMETHEUS_API_HOST", prometheusEndpoint)
 	if err != nil {
 		return false, err
 	}
 	hasChanged = hasChanged || changed
 
 	// Prometheus host end point ssl verify
 	prometheusEndpointSSLVerify := strconv.FormatBool(c.spec.Dashboard.PrometheusEndpointSSLVerify)
-	changed, err = monStore.SetIfChanged(daemonID, "mgr/dashboard/PROMETHEUS_API_SSL_VERIFY", prometheusEndpointSSLVerify)
+	changed, err = monStore.SetIfChanged(config.MgrType, "mgr/dashboard/PROMETHEUS_API_SSL_VERIFY", prometheusEndpointSSLVerify)
 	if err != nil {
 		return false, err
 	}
 	hasChanged = hasChanged || changed
 
 	// server port
 	port := strconv.Itoa(c.dashboardInternalPort())
-	changed, err = monStore.SetIfChanged(daemonID, "mgr/dashboard/server_port", port)
+	changed, err = monStore.SetIfChanged(config.MgrType, "mgr/dashboard/server_port", port)
 	if err != nil {
 		return false, err
 	}
 	hasChanged = hasChanged || changed
 
 	// SSL enabled. Needed to set specifically the ssl port setting
 	if c.spec.Dashboard.SSL {
-		changed, err = monStore.SetIfChanged(daemonID, "mgr/dashboard/ssl_server_port", port)
+		changed, err = monStore.SetIfChanged(config.MgrType, "mgr/dashboard/ssl_server_port", port)
 		if err != nil {
 			return false, err
 		}
 		hasChanged = hasChanged || changed
 	}
 
+	// Remove any existing per mgr-daemon configuration
+	if removeMgrDaemonConfiguration {
+		logger.Info("Removing any previous per mgr-daemon configuration")
+		for _, daemonID := range c.getDaemonIDs() {
+			mgrDaemonID := fmt.Sprintf("%s.%s", config.MgrType, daemonID)
+			monStore.Delete(mgrDaemonID, "mgr/dashboard/url_prefix")
+			monStore.Delete(mgrDaemonID, "mgr/dashboard/ssl")
+			monStore.Delete(mgrDaemonID, "mgr/dashboard/PROMETHEUS_API_HOST")
+			monStore.Delete(mgrDaemonID, "mgr/dashboard/PROMETHEUS_API_SSL_VERIFY")
+			monStore.Delete(mgrDaemonID, "mgr/dashboard/server_port")
+			monStore.Delete(mgrDaemonID, "mgr/dashboard/ssl_server_port")
+		}
+		removeMgrDaemonConfiguration = false
+	}
+
```

Comment:
At what version of Ceph does this need to be done?

Presumably, the changes in Ceph's code made these redundant at a specific version (or versions), and we should make sure we don't accidentally break older verisons by doing this.

Reply:
@BlaineEXE ceph dashboard (in cephadm based deployments) has always used configuration at mgr level. Not sure why in Rook we were using per-daemon config. This shouldn't be an issue but if for any strange reason there's a mismatch in that conf we could end up in weird issues. 

With this change I'm just removing the usage of per-daemon configs and moving to using the generic mgr section for all the dashboard related conf parameters. 

Does this answer your question?

Reply:
Got it! Yes, I missed the generic vs. specific subtext here.